cmakemodules/script_pcl.cmake

```diff
@@ -14,10 +14,11 @@ IF(NOT DISABLE_PCL)
 	# PCL library:
 	# --------------------------------------------
 	find_package(PCL COMPONENTS io common registration visualization segmentation surface QUIET)
-	if (PCL_FOUND)
+	if(PCL_FOUND AND PCL_IO_FOUND AND PCL_COMMON_FOUND AND PCL_REGISTRATION_FOUND AND PCL_VISUALIZATION_FOUND AND PCL_SURFACE_FOUND)
+
 		# Filter: It seems clang + c++11 fails to build against PCL (for clang <3.5):
 		if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" AND NOT CMAKE_MRPT_CLANG_VERSION GREATER 34)
```

Comment:
You mean those  SEGFAULTs?

Since GCC 6 c++14 is default, so I'm not too worried about the future. I feel like most of those bugs have gone away in 16.04. I recall having serious issues with 14.04. The issue was binary compatibility with boost[1], not PCL.

Adding PCL to 16.04 requires repackaging of VTK and PCL anyway, so it's unlikely to affect many people. Hopefully VTK can get it right in 18.04, so we can finally have a good PCL package in the Ubuntu repo.

[1] https://lists.boost.org/Archives/boost/2013/05/203762.php